File: fsharp/bob/Bob.fs
module Bob

open System.Text.RegularExpressions

let IsYell (input: string): bool =
  input = input.ToUpper() && Regex.IsMatch(input, "[A-Z]")

let IsQuestion (input: string): bool =
  input.EndsWith("?")

let IsNothing (input: string): bool = 
  input.Trim() = ""

let hey (input: string): string =
  if IsYell(input) then "Whoa, chill out!"
  else if IsQuestion(input) then "Sure."
  else if IsNothing(input) then "Fine. Be that way!"
  else "Whatever."

File: csharp/bob/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Hey(string statement)
    {
        statement = statement.Trim();

        if (IsNothing(statement))
            return "Fine. Be that way!";

        if (IsYelling(statement))
            return "Whoa, chill out!";

        if (isQuestion(statement))
            return "Sure.";

        return "Whatever.";

    }

    private static bool IsYelling(string statement)
    {
        if (!new Regex("[a-zA-Z]").IsMatch(statement))
            return false;

        if (statement != statement.ToUpper())
            return false;

        return true;
    }

    private static bool isQuestion(string statement)
    {
        return statement.EndsWith("?");
    }

    private static bool IsNothing(string statement)
    {
        return statement?.Length == 0;
    }
}